library/azure_rm_applicationsecuritygroup.py

```diff
@@ -0,0 +1,250 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018 Yunge Zhu, <yungez@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_applicationsecuritygroup
+version_added: "2.8"
+short_description: Manage Azure Application Security Group.
+description:
+    - Create, update and delete instance of Azure Application Security Group.
+
+options:
+    resource_group:
+        description:
+            - The name of the resource group.
+        required: True
+    name:
+        description:
+            - The name of the application security group.
+        required: True
+    location:
+        description:
+            - Resource location. If not set, location from the resource group will be used as default.
+    state:
+      description:
+        - Assert the state of the Application Security Group.
+        - Use 'present' to create or update an Application Security Group and 'absent' to delete it.
+      default: present
+      choices:
+        - absent
+        - present
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Yunge Zhu (@yungezz)"
+
+'''
+
+EXAMPLES = '''
+  - name: Create application security group
+    azure_rm_applicationsecuritygroup:
+      resource_group: myRG
+      name: MyAsg
+      location: eastus
+      tags:
+        foo: bar
+'''
+
+RETURN = '''
+id:
+    description:
+        - Resource id of the application security group.
+    returned: always
+    type: str
+    sample: /subscriptions/<subid>/resourceGroups/rg1/providers/Microsoft.Network/applicationSecurityGroups/MyAsg
```

Comment:
just for consistency with passed parameters in examples, you could replace:
rg1 -> myRG

Reply:
fixed